src/main/java/NestedNullRefactoring.java

```diff
@@ -0,0 +1,155 @@
+ * methods which return the nullness of a value are replaced with the method's
+ * one-line null check. Preserves semantics by copying the invoked code
+ * precisely.
+ */
+public class NestedNullRefactoring extends Refactoring {
+	public static final String NAME = "NestedNullRefactoring";
+
+	private final Dictionary<IMethodBinding, Expression> applicableMethods;
+
+	public NestedNullRefactoring() {
+		applicableMethods = new Hashtable<>();
+	}
+
+	@Override
+	public boolean isApplicable(ASTNode node) {
+		if (node instanceof MethodInvocation invocation) {
+			return isApplicable(invocation);
+		}
+
+		if (node instanceof MethodDeclaration declaration) {
+			return isApplicable(declaration);
+		}
+
+		return false;
+	}
+
+	/*
+	 * Returns true iff the provided invocation is of a registered one-line method
+	 * that returns the result of a null check
+	 */
+	public boolean isApplicable(MethodInvocation invocation) {
+		if (applicableMethods.get(invocation.resolveMethodBinding()) != null) {
+			System.out.println("[DEBUG] Invocation of applicable method found");
+			return true;
+		}
+		return false;
+	}
+
+	/*
+	 * Returns true iff Node is a one-line private or package-private method that
+	 * returns the result of a null check
+	 */
+	public boolean isApplicable(MethodDeclaration declaration) {
+		Type retType = declaration.getReturnType2();
+		boolean returnsBoolean = (retType.isPrimitiveType()
+				&& ((PrimitiveType) retType).getPrimitiveTypeCode() == PrimitiveType.BOOLEAN);
+		if (!(returnsBoolean)) {
+			return false;
+		}
+
+		// Ensure the method declaration is private
+		if (!((declaration.getModifiers() & Modifier.PRIVATE) == Modifier.PRIVATE)) {
+			return false;
+		}
+
+		// Checks if there are any parameters
+		// TODO: Make work with parameters in case of methods that take variables to
+		// check as inputs, such as `checkObject(Object o) { return o !=null }}`
+		boolean hasParams = declaration.parameters().size() > 0;
+		if (hasParams) {
+			return false;
+		}
+
+		Block body = declaration.getBody();
+		List<Statement> stmts = body.statements();
+
+		boolean isOneLine = stmts.size() == 1;
+		if (!isOneLine) {
+			return false;
+		}
+
+		// Possible unneccessary since we already confirmed return type is not void
+		Statement stmt = stmts.get(0);
+		if (!(stmt instanceof ReturnStatement)) {
+			return false;
+		}
+
+		// Checks that the return statement is of a single equality check
+		Expression retExpr = ((ReturnStatement) stmt).getExpression();
+		if (!(retExpr instanceof InfixExpression)) {
+			return false;
+		}
+
+		InfixExpression infix = (InfixExpression) retExpr;
+
+		if ((infix.getOperator() == InfixExpression.Operator.NOT_EQUALS
+				|| infix.getOperator() == InfixExpression.Operator.EQUALS)) {
+			Expression leftOperand = infix.getLeftOperand();
+			Expression rightOperand = infix.getRightOperand();
+
+			if ((leftOperand instanceof SimpleName && rightOperand instanceof NullLiteral)
```

Comment:
Why do we only support simple names? For example, I would expect this refactoring to also apply for this method:
```
boolean checkFieldIsNotNull() {
    return this.field != null;
}
```
I think right now the "this" will prevent the refactoring from firing. Can you add the associated test case and implement this, as well?